cli/azd/cmd/auth_login.go

```diff
@@ -292,6 +300,27 @@ func countTrue(elms ...bool) int {
 	return i
 }
 
+// runningOnCodespacesBrowser use `code --status` which returns:
+//
+//	> The --status argument is not yet supported in browsers.
+//
+// to detect when vscode is within a WebBrowser environment.
+func runningOnCodespacesBrowser(ctx context.Context, commandRunner exec.CommandRunner) bool {
+	runArgs := exec.NewRunArgs("code", "--status")
+	result, err := commandRunner.Run(ctx, runArgs)
+	if err != nil {
+		// An error here means VSCode is not installed or found, or something else.
+		// At any case, we know VSCode is not within a webBrowser
+		return false
+	}
+
+	if strings.Contains(result.Stdout, "The --status argument is not yet supported in browsers") {
+		return true
+	}
+
+	return false
```

Comment:
nit: can go either way

```suggestion
	return strings.Contains(result.Stdout, "The --status argument is not yet supported in browsers")
```

Reply:
ah!  nice catch!!  thank you!